src/led.c

```diff
@@ -169,11 +170,13 @@ enum bl_error bl_led_set(uint16_t led_mask)
 void bl_led_status_set(bool enable)
 {
 #if (BL_REVISION >= 2)
+	const struct device * gpio;
+	gpio = device_get_binding(DT_LABEL(DT_NODELABEL(GPIOB)));
 	if (enable) {
-		gpio_mode_setup(GPIOB, GPIO_MODE_OUTPUT, GPIO_PUPD_NONE, GPIO7);
-		gpio_clear(GPIOB, GPIO7);
+		gpio_pin_configure(gpio, 7, GPIO_OUTPUT);
+		gpio_port_clear_bits(GPIOB, GPIO7);
```

Comment:
Looks like `gpio_port_clear_bits` takes a `device*` too. So you'll also need to pass it `gpio` instead of `GPIOB`